Source/WebCore/page/scrolling/ScrollAnchoringController.h

```diff
@@ -32,20 +32,31 @@ namespace WebCore {
 
 class Element;
 
+enum class CandidateExaminationResult {
+    Exclude, Select, Descend, Skip
+};
+
 class ScrollAnchoringController final : public CanMakeWeakPtr<ScrollAnchoringController> {
     WTF_MAKE_FAST_ALLOCATED;
 public:
     explicit ScrollAnchoringController(LocalFrameView& frameView)
         : m_frameView(frameView)
     { }
+
     void invalidateAnchorElement();
     void adjustScrollPositionForAnchoring();
     void selectAnchorElement();
-    void chooseAnchorElement();
+    void chooseAnchorElement(Document&);
     void updateAnchorElement();
     LocalFrameView& frameView() { return m_frameView; }
 
 private:
+    Element* findAnchorElementRecursive(Element*);
+    CandidateExaminationResult examineCandidate(Element&);
+    bool examinePriorityCandidate(Element*);
```

Comment:
You should name the functions to make it clear what the bool return values mean.

Can these be static?

Reply:
Changed these to static.